src/browser/atom-window.coffee

```diff
@@ -73,7 +73,8 @@ class AtomWindow
     @browserWindow.loadUrl @getUrl(loadSettings)
     @browserWindow.focusOnWebView() if @isSpec
 
-    @openLocations(locationsToOpen) unless @isSpecWindow()
+    hasPathToOpen = not (locationsToOpen.length is 1 and not locationsToOpen[0].pathToOpen?)
```

Comment:
I'd rather have the check outside `openLocations` for clarity when scanning this code. openLocations can handle the null case just fine (it opens an empty editor). 

The empty editor logic is moving to the renderer process.